xbmc/favourites/GUIWindowFavourites.cpp

```diff
@@ -145,12 +149,12 @@ bool CGUIWindowFavourites::OnAction(const CAction& action)
 
   if (action.GetID() == ACTION_PLAYER_PLAY)
   {
-    const CFavouritesURL favURL((*m_vecItems)[selectedItem]->GetPath());
-    if (!favURL.IsValid())
+    const auto target{
+        CServiceBroker::GetFavouritesService().ResolveFavourite(*(*m_vecItems)[selectedItem])};
+    if (!target)
       return false;
 
-    CFileItem item{favURL.GetTarget(), favURL.IsDir()};
-    item.LoadDetails();
+    CFileItem item{*target};
```

Comment:
Same here, is the copy needed?

Reply:
Maybe not, was aking myself the same, then simply took over the pattern from existing code. Will put to check this on my todo list for later.